api/v1beta1/azuremanagedcontrolplane_default_test.go

```diff
@@ -53,3 +54,36 @@ func hardcodedAzureManagedControlPlaneWithSSHKey(sshPublicKey string) *AzureMana
 		},
 	}
 }
+
+func TestSetDefaultAutoScalerProfile(t *testing.T) {
+	m := &AzureManagedControlPlane{
+		Spec: AzureManagedControlPlaneSpec{
+			AutoScalerProfile: &AutoScalerProfile{},
+		},
+	}
+
+	// Test when all fields are nil
+	m.setDefaultAutoScalerProfile()
+
+	g := NewWithT(t)
+
+	// Verify that the default values are set
```

Comment:
should we also add a test case to verify that custom values don't get overwritten?

Reply:
Agreed! Good catch

Reply:
Should i add that test checking for all values or for some values?

Reply:
I would shoot for all values. I think that we can test passing in an autoscaler with all values populated by something other than the default, then ensuring the output matches.

Reply:
I have made the changes. Please review.